gammapy/utils/fitting/sampling.py

```diff
@@ -0,0 +1,162 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# helper functions for mcmc smapling
+import numpy as np
+import emcee
+import matplotlib.pyplot as plt
+import corner
+
+__all__ = ["uniform_prior", "run_mcmc", "plot_trace", "plot_corner"]
+
+#TODO: so far only works with a uniform prior on parameters
+
+# Prior functions
+def uniform_prior(value, umin, umax):
+    """Uniform prior distribution."""
+    if umin <= value <= umax:
+        return 0.0
+    else:
+        return -np.inf
+
+
+def normal_prior(value, mean, sigma):
+    """Normal prior distribution."""
+    return -0.5 * (2 * np.pi * sigma) - (value - mean) ** 2 / (2.0 * sigma)
+
+
+# Read/write parameters in the dataset
+def model_to_par(dataset):
```

Comment:
Maybe `dataset_free_parameter_factors` to be more explicit?

Reply:
done